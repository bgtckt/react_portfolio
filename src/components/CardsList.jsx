import React from 'react';
import CardItem from "./CardItem";

const CardsList = ({items, setItems}) => {
  // функция для удаления карточки
  function removeCard (card) {
    setItems(items.filter(elem => elem.id !== card.id))
  }

  // проверка на наличие карточек в массиве и вывод сообщения в случае их отсутствия
  if (items.length === 0) {
    return (
      <h3 className='clients__error'>Список клиентов пуст!</h3>
    )
  }

  return (
    <div className='clients__list' data-testid='cards-list'>
      {items.map(card =>
        <CardItem
          card={card}
          key={card.id}
          remove={removeCard}
        />
      )}
    </div>
  );
};

export default CardsList;