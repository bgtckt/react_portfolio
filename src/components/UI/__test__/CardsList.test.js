import {render, screen} from '@testing-library/react';
import CardsList from "../../CardsList";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import userEvent from "@testing-library/user-event/dist";

const data = [
  {
    id: 1,
    name: "Leanne Graham",
    email: "Sincere@april.biz",
    phone: "1-770-736-8031 x56442",
    company: {
      name: "Romaguera-Crona"
    }
  },
  {
    id: 2,
    name: "Ervin Howell",
    email: "Shanna@melissa.tv",
    phone: "010-692-6593 x09125",
    company: {
      name: "Deckow-Crist"
    }
  }
];

describe('CardsList component', () => {
  it ('CardsList renders', () => {
    render(
      <BrowserRouter>
        <CardsList items={data}/>
      </BrowserRouter>
    );
    expect(screen.getByTestId('cards-list')).toBeInTheDocument();
  });

  it ('CardsList not renders without data', () => {
    render(
      <BrowserRouter>
        <CardsList items={[]}/>
      </BrowserRouter>
    );
    expect(screen.getByText('Список клиентов пуст!')).toBeInTheDocument();
    expect(screen.queryByTestId('cards-list')).toBeNull();
  });

  it ('CardsList snapshot', () => {
    const cardsList = render(
      <BrowserRouter>
        <CardsList items={data}/>
      </BrowserRouter>
    );
    expect(cardsList).toMatchSnapshot();
  });
});