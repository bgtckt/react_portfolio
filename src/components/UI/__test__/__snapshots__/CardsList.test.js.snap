// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList component CardsList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="clients__list"
        data-testid="cards-list"
      >
        <div
          class="clients__card"
          data-testid="card-item"
        >
          <h4
            class="clients__name"
          >
            1
            . 
            Leanne Graham
          </h4>
          <div
            class="clients__info"
          >
            <p>
              <b>
                email:
              </b>
               
              Sincere@april.biz
            </p>
            <p>
              <b>
                phone:
              </b>
               
              1-770-736-8031 x56442
            </p>
            <p>
              <b>
                company:
              </b>
               "
              Romaguera-Crona
              "
            </p>
          </div>
          <button
            class="btn"
          >
            Удалить
          </button>
        </div>
        <div
          class="clients__card"
          data-testid="card-item"
        >
          <h4
            class="clients__name"
          >
            2
            . 
            Ervin Howell
          </h4>
          <div
            class="clients__info"
          >
            <p>
              <b>
                email:
              </b>
               
              Shanna@melissa.tv
            </p>
            <p>
              <b>
                phone:
              </b>
               
              010-692-6593 x09125
            </p>
            <p>
              <b>
                company:
              </b>
               "
              Deckow-Crist
              "
            </p>
          </div>
          <button
            class="btn"
          >
            Удалить
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="clients__list"
      data-testid="cards-list"
    >
      <div
        class="clients__card"
        data-testid="card-item"
      >
        <h4
          class="clients__name"
        >
          1
          . 
          Leanne Graham
        </h4>
        <div
          class="clients__info"
        >
          <p>
            <b>
              email:
            </b>
             
            Sincere@april.biz
          </p>
          <p>
            <b>
              phone:
            </b>
             
            1-770-736-8031 x56442
          </p>
          <p>
            <b>
              company:
            </b>
             "
            Romaguera-Crona
            "
          </p>
        </div>
        <button
          class="btn"
        >
          Удалить
        </button>
      </div>
      <div
        class="clients__card"
        data-testid="card-item"
      >
        <h4
          class="clients__name"
        >
          2
          . 
          Ervin Howell
        </h4>
        <div
          class="clients__info"
        >
          <p>
            <b>
              email:
            </b>
             
            Shanna@melissa.tv
          </p>
          <p>
            <b>
              phone:
            </b>
             
            010-692-6593 x09125
          </p>
          <p>
            <b>
              company:
            </b>
             "
            Deckow-Crist
            "
          </p>
        </div>
        <button
          class="btn"
        >
          Удалить
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
